src/numbers/xdrint.js

```diff
@@ -0,0 +1,246 @@
+/* eslint no-bitwise: ["error", {"allow": [">>"]}] */
+import { Hyper, UnsignedHyper } from 'js-xdr';
+
+import { Uint128 } from './uint128';
+import { Uint256 } from './uint256';
+import { Int128 } from './int128';
+import { Int256 } from './int256';
+
+import xdr from '../xdr';
+
+export class XdrInt {
+  int; // child class of a jsXdr.LargeInt
+  type; // string, one of i64, u64, i128, u128, i256, or u256
+
+  /**
+   * Transforms an opaque {@link xdr.ScVal} into a native BigInt, if possible.
+   *
+   * You can then give this back to create an {@link ScInt} instance, but the
+   * rationale here is that the native type is more likely to be immediately
+   * useful.
+   *
+   * @param {xdr.ScVal} scv - the raw XDR value to parse into an integer
+   * @returns {bigint} the integer value, regardless of size (even 32-bit)
```

Comment:
Shouldn't `XdrInt.fromScVal(scv)` return an `XdrInt`?

Reply:
Seems like this could just do the switch to determine the `type`, then call into the constructor...

Reply:
You're right that this can be refactored now that `XdrInt` exists! I did that, and also moved it outside of the class and renamed it to `convertScVal` to be more explicit about its purpose.

The reason why it doesn't return `XdrInt` is to support 32-bit ints. It's not perfect, I know, but because of https://github.com/stellar/js-stellar-base/pull/620#discussion_r1223408413, we can't wrap it all up into one interface.